pkg/spanconfig/spanconfigkvsubscriber/kvsubscriber.go

```diff
@@ -22,10 +22,19 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/spanconfig"
 	"github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigstore"
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
+	"github.com/cockroachdb/cockroach/pkg/util/metric"
 	"github.com/cockroachdb/cockroach/pkg/util/stop"
 	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
 )
 
+var durationSinceLastPartialUpdate = metric.Metadata{
```

Comment:
This metric isn't quite the "duration since the partial update", right? We're updating the gauge every time we receive an update, so it's more "latency difference for the last update we received" I think. If we're stuck (job not running, closed ts not bumping), the last recorded value would be the delta between the wall time when we last received the value and the timestamp of the update itself. I think this metric is still very useful, if stuck we'll see it flatline and then when finally unstuck, see a large increase. The documentation + naming here and below should be reworded to reflect the _specific_ way to interpret this metric. A parallel is "kv.closed_timestamp.max_behind_nanos". Also below, the Measurement field should say "Nanoseconds".

Reply:
Discussed offline, the suggestion does give a better signal so I've updated the diff.